### PR TITLE
Mat 6819 multi populations work sheet

### PR DESCRIPTION
**Pull request by sb-cecilialiu**
## MADiE PR

Jira Ticket: [MAT-6819](https://jira.cms.gov/browse/MAT-6819)
(Optional) Related Tickets:

### All Submissions

- [x] This PR has the JIRA linked.
- [x] Required tests are included.
- [x] No extemporaneous files are included (i.e Complied files or testing results).
- [x] This PR is merging into the **correct branch**.
- [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
- [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps

If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

- [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).
- [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo).

### Reviewers

By Approving this PR you are attesting to the following:

- Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
- The tests appropriately test the new code, including edge cases.
- If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/auth/auth.guard.spec.ts [hunk 1]
```diff
@@ -48,7 +60,7 @@ describe('AuthGuard', () => {
...
+              'Bearer eyJraWQiOiJNNG9CMW9DSmthdC0tYTNENFFXUFA3RWZCbUl3NG9BV05KYWJxdEJhUnM4IiwiYWxnIjoiUlMyNTYifQ.eyJ2ZXIiOjEsImp0aSI6IkFULmRQdFdGMndrYlN5VjJxVW1BanQ0dXBERklwOExMT2poRFl4MWxzYWJkaWciLCJpc3MiOiJodHRwczovL2Rldi0xODA5MjU3OC5va3RhLmNvbS9vYXV0aDIvZGVmYXVsdCIsImF1ZCI6ImFwaTovL2RlZmF1bHQiLCJpYXQiOjE3MTI5MjYyNTIsImV4cCI6MTcxMjkyOTg1MiwiY2lkIjoiMG9hMmZxdGF6OTVmcUpxYmY1ZDciLCJ1aWQiOiIwMHUzaTNjM3p6WlhLcjkwMTVkNyIsInNjcCI6WyJwcm9maWxlIiwib3BlbmlkIl0sImF1dGhfdGltZSI6MTcxMjkyNjI1MSwic3ViIjoiY2VjaWxpYS5saXVAc2VtYW50aWNiaXRzLmNvbSJ9.bdZh7ygtJKvausHo58VGPLonPeWPAGXDFXHDY445ERxwCpNSiy-VdfZF3TZfg_s9k-XAwRGcd-T7lcy4dW9Qh6L74bdDucYX_TSR8WreLvS3W4o-uyRi0HQU8lB2axF2xqdB_UGEr79Am-TZhpQmwIQWXPTUpMr33fxun5ic6Tidq8WFgVVIBXTusKUHopaIP8z4gzapPo6hxLtJ6farGZcC1URHMk2o6DeyGuvO4i4y8vpowwqIHKLZ55P9kLgtbehC4LYPaRqoqWzngjF3Zy_4StmdxaFM2U_xulO5L3LysporElvVlU11HQAluXCff_PYYeV25u1Xbnp3_-W34Q',
```
Comment (adongare) [inline src/auth/auth.guard.spec.ts]: can we not have some mock sting? gitleaks is flagging this

Comment (sb-cecilialiu) [reply src/auth/auth.guard.spec.ts]: @adongare I think it's flagging line 45:               sub: 'aUser@company.com',

File: src/services/export.service.ts [hunk 2]
```diff
@@ -191,41 +219,53 @@ export class ExportService {
...
+    populationDto: PopulationDto,
+    result,
+  ) {
+    let foundPopulation: PopulationDto = null;
+    result.populations?.forEach((currentPopulation) => {
+      if (currentPopulation.name === populationDto.name) {
+        foundPopulation = currentPopulation;
+      }
+    });
+    testCaseData.push(foundPopulation?.actual);
```
Comment (adongare) [inline src/services/export.service.ts]: can be optimized 
```suggestion
    const population = result.populations?.find((currentPopulation) => currentPopulation.name === populationDto.name);
testCaseData.push(population?.actual);
```

File: src/services/export.service.ts [hunk 3]
```diff
@@ -191,41 +219,53 @@ export class ExportService {
...
+  private getPopulations = (testCaseExcelExportDto: TestCaseExcelExportDto) => {
+    let populations: PopulationDto[] = [];
+    testCaseExcelExportDto.testCaseExecutionResults?.forEach(
+      (result: TestCaseExecutionResultDto) => {
+        if (result.populations?.length > 0) {
+          populations = result.populations;
+        }
+      },
+    );
+    return populations;
```
Comment (adongare) [inline src/services/export.service.ts]: I'm lil bit confused about this loop. Are you trying to get populations from any one of the executionResults? because this loop will iterate even after you fulfill `result.populations?.length > 0`. there is no breaking once you get the populations and it keeps looping and overriding `populations = result.populations;`. if you just want populations from any execution result, then:
```suggestion
    const result  = testCaseExcelExportDto.testCaseExecutionResults?.find(result) => result.populations?.length > 0);
    return result? result.populations : [];
```

File: src/auth/auth.guard.spec.ts (hunk 1 above)
Comment (adongare) [reply src/auth/auth.guard.spec.ts]: oh yeah. maybe we could just enter a simple string instead of email and even for this token unless it is required by test authguard.

File: src/services/export.service.ts (hunk 2 above)
Comment (sb-cecilialiu) [reply src/services/export.service.ts]: @adongare Changed.

File: src/services/export.service.ts (hunk 3 above)
Comment (sb-cecilialiu) [reply src/services/export.service.ts]: @adongare Changed

**Review by adongare — APPROVED** [review]
